source/mir/bignum/integer.d

```diff
@@ -317,6 +318,116 @@ struct BigInt(size_t maxSize64)
         return overflow;
     }
 
+    /++
+    Performs division & extracts the remainder.
+    Params:
+        rhs = unsigned value to divide by
+    Returns:
+        unsigned remainder value
+    +/ 
+    auto divRem(size_t rhsSize)(UInt!rhsSize rhs)
+        @safe pure nothrow @nogc scope
+    {
+        import mir.bignum.low_level_view : divm, BigUIntView;
+
+        enum m = ((data.length * (size_t.sizeof * 8)) / (uint.sizeof * 8)), n = (rhsSize / (uint.sizeof * 8));
+
+        struct QuotientAndRemainder
+        {
+            // XXX: we have to assume the quotient size here,
+            // checking at runtime would be more accurate for better space savings
+            UInt!(data.length * (size_t.sizeof * 8)) quotient;
+            UInt!rhsSize remainder;
+        }
+
+        QuotientAndRemainder val;
+
+        uint[m + 1] _div = void;
+        // shouldn't be unaligned here -- maybe a possibility though
+        _div[$ - 1] = 0;
+        _div[0 .. $ - 1] = (cast(uint[])data);
+
+        BigUIntView!uint dividend = BigUIntView!uint(_div);
+        BigUIntView!uint divisor = cast(BigUIntView!uint)rhs.view;
+        BigUIntView!uint quotient = cast(BigUIntView!uint)val.quotient.view;
+        divm(dividend, divisor, quotient);
+
+        BigUIntView!uint remainder = cast(BigUIntView!uint)val.remainder.view;
+        remainder.coefficients[0 .. $] = (cast(uint[])_div)[0 .. remainder.coefficients.length];
+
+        return val;
+    }
+
+    /++
+    Performs `big /= rhs` operation.
+    Params:
+        rhs = unsigned value to divide by
+    Returns:
+        quotient from division
+    +/
+    ref opOpAssign(string op : "/", size_t size)(UInt!size rhs)
```

Comment:
Also, `opOpAssign` should accept `ref const BigInt!..` and `long` as well.

Reply:
Ah... you want me to implement signed division now. I'll have to do more research on how to adapt our code to do this - Knuth's Algorithm D is specifically meant for unsigned division.

Reply:
yep, we don't need to touch the algorithm! just play with signs inside BigInt methods.  That is enough .